### ISSUE TITLE
no-unused-prop-types doesn't report errors on readonly props

### ISSUE THREAD

**Issue by lukeapage** [yannickcr/eslint-plugin-react#2358]
If you wrap the props in $ReadOnly<> then no-unused-prop-types doesn't validate.

```
type Props = $ReadOnly<{| 
    a: string,
 |}>;
```

**Comment by ljharb** [comment]
Is this typescript or flow? In flow, wouldn't you use `+` to indicate that the prop is readonly?

**Comment by lukeapage** [comment]
It’s flow. And yes you can use + but it’s less work to wrap props in the helper.

Also for fixing 1200 occurrences in an existing codebase I could insert the read only utility with a regex a bit easier, and that’s how I found this. I can look into fixing it.
